lib/compiler/src/v3_kernel.erl

```diff
@@ -2141,21 +2155,7 @@ arg_arg(Con) -> Con.
 arg_alias(#ialias{vars=As}) -> As;
 arg_alias(_Con) -> [].
 
-arg_con(Arg) ->
-    case arg_arg(Arg) of
-	#k_literal{} -> k_literal;
-	#k_int{} -> k_int;
-	#k_float{} -> k_float;
-	#k_atom{} -> k_atom;
-	#k_nil{} -> k_nil;
-	#k_cons{} -> k_cons; 
-	#k_tuple{} -> k_tuple;
-	#k_map{} -> k_map;
-	#k_binary{} -> k_binary;
-	#k_bin_end{} -> k_bin_end;
-	#k_bin_seg{} -> k_bin_seg;
-	#k_var{} -> k_var
-    end.
+arg_con(Arg) -> element(1, arg_arg(Arg)).
```

Comment:
IMO this change is a code degradation.  The original code contains perfect type information to both the input argument and the result of the function while the new code is basically "anything goes".  Most notably, `#ialias{}` is now also handled (and probably in an unintended way) when in the original code it was not.

In general, I would advice _against_ such "optimizations"/code rewrites.   They are typically a very bad idea that come back to bite you at some point.

Reply:
Yup. I was 80% sure that this change would be rejected. But I thought I would submit it to review anyway. :) I will revert it.